File: matete-web/src/Entity/Annonce.php
<?php

namespace App\Entity;

use App\Repository\AnnonceRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;

/**
 * @ORM\Entity(repositoryClass=AnnonceRepository::class)
 */
class Annonce
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     */
    private $id;

    /**
     * @ORM\Column(type="datetime_immutable")
     */
    private $creneauxDebut;

    /**
     * @ORM\Column(type="datetime_immutable")
     */
    private $creneauxFin;

    /**
     * @ORM\Column(type="string", length=50)
     */
    private $libelleProduit;

    /**
     * @ORM\Column(type="float")
     */
    private $prixUnitaire;

    /**
     * @ORM\Column(type="integer")
     */
    private $quantite;

    /**
     * @ORM\ManyToOne(targetEntity=Lieu::class, inversedBy="Annonce")
     */
    private $lieu;

    /**
     * @ORM\ManyToOne(targetEntity=Categorie::class, inversedBy="Annonce")
     */
    private $categorie;

    /**
     * @ORM\ManyToMany(targetEntity=Commande::class, mappedBy="Annonce")
     */
    private $commandes;

    /**
     * @ORM\ManyToOne(targetEntity=Producteur::class, inversedBy="Annonce")
     */
    private $producteur;

    public function __construct()
    {
        $this->commandes = new ArrayCollection();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getCreneauxDebut(): ?\DateTimeImmutable
    {
        return $this->creneauxDebut;
    }

    public function setCreneauxDebut(\DateTimeImmutable $creneauxDebut): self
    {
        $this->creneauxDebut = $creneauxDebut;

        return $this;
    }

    public function getCreneauxFin(): ?\DateTimeImmutable
    {
        return $this->creneauxFin;
    }

    public function setCreneauxFin(\DateTimeImmutable $creneauxFin): self
    {
        $this->creneauxFin = $creneauxFin;

        return $this;
    }

    public function getLibelleProduit(): ?string
    {
        return $this->libelleProduit;
    }

    public function setLibelleProduit(string $libelleProduit): self
    {
        $this->libelleProduit = $libelleProduit;

        return $this;
    }

    public function getPrixUnitaire(): ?float
    {
        return $this->prixUnitaire;
    }

    public function setPrixUnitaire(float $prixUnitaire): self
    {
        $this->prixUnitaire = $prixUnitaire;

        return $this;
    }

    public function getQuantite(): ?int
    {
        return $this->quantite;
    }

    public function setQuantite(int $quantite): self
    {
        $this->quantite = $quantite;

        return $this;
    }

    public function getLieu(): ?Lieu
    {
        return $this->lieu;
    }

    public function setLieu(?Lieu $lieu): self
    {
        $this->lieu = $lieu;

        return $this;
    }

    public function getCategorie(): ?Categorie
    {
        return $this->categorie;
    }

    public function setCategorie(?Categorie $categorie): self
    {
        $this->categorie = $categorie;

        return $this;
    }

    /**
     * @return Collection|Commande[]
     */
    public function getCommandes(): Collection
    {
        return $this->commandes;
    }

    public function addCommande(Commande $commande): self
    {
        if (!$this->commandes->contains($commande)) {
            $this->commandes[] = $commande;
            $commande->addAnnonce($this);
        }

        return $this;
    }

    public function removeCommande(Commande $commande): self
    {
        if ($this->commandes->removeElement($commande)) {
            $commande->removeAnnonce($this);
        }

        return $this;
    }

    public function getProducteur(): ?Producteur
    {
        return $this->producteur;
    }

    public function setProducteur(?Producteur $producteur): self
    {
        $this->producteur = $producteur;

        return $this;
    }
}

File: Android(Matetest)/app/src/main/java/com/example/matetest/Annonce.java
package com.example.matetest;

public class Annonce {

    private int id;
    private String creneaux_debut;
    private String creneaux_fin;
    private String libelle_produit;
    private Double prix_unitaire;
    private int quantite;
    private String status;
    private Lieu unlieu;
    private Categorie unecategorie;
    private Producteur unproducteur;
    private boolean isSelected;


    public Annonce(int id, String creneaux_debut,String creneaux_fin,String libelle_produit, double prix_unitaire,int quantite,String status,Lieu unlieu,Categorie unecategorie,Producteur unproducteur){
        this.id = id;
        this.creneaux_debut = creneaux_debut;
        this.creneaux_fin = creneaux_fin;
        this.libelle_produit = libelle_produit;
        this.prix_unitaire = prix_unitaire;
        this.quantite = quantite;
        this.status = status;
        this.unlieu = unlieu;
        this.unecategorie = unecategorie;
        this.unproducteur = unproducteur;

    }

    public int getId(){
        return id;
    }
    public String getCreneaux_debut(){
        return creneaux_debut;
    }
    public String getCreneaux_fin(){
        return creneaux_fin;
    }
    public String getLibelle_produit(){
        return libelle_produit;
    }
    public Double getPrix_unitaire(){
        return prix_unitaire;
    }
    public int getQuantite(){
        return quantite;
    }
    public String getStatus(){
        return status;
    }
    public Lieu getLieu(){
        return unlieu;
    }
    public Categorie getCategorie(){
        return unecategorie;
    }
    public Producteur getProducteur(){
        return unproducteur;
    }


    public boolean isSelected() {
        return isSelected;
    }

    public void setSelected(boolean selected) {
        isSelected = selected;
    }

    @Override
    public String toString() {
        return "Annonce{" +
                "id=" + id +
                ", creneaux_debut='" + creneaux_debut + '\'' +
                ", creneaux_fin='" + creneaux_fin + '\'' +
                ", libelle_produit='" + libelle_produit + '\'' +
                ", prix_unitaire=" + prix_unitaire +
                ", quantite=" + quantite +
                ", status='" + status + '\'' +
                ", unlieu=" + unlieu +
                ", unecategorie=" + unecategorie +
                ", isSelected=" + isSelected +
                ", unproducteur=" + unproducteur +
                '}';
    }
}